source/docs/prismic/routable-types.md

```diff
@@ -0,0 +1,231 @@
+---
+id: routable-types
+title: Prismic Routable Types
+---
+
+To make a custom prismic data type routable these steps are required:
+
+- [Create a field to represent the Url of the custom type](#create-a-field-to-represent-the-url-of-the-custom-type)
+
+- [Create a GraphQL type to represent the custom data type](#create-a-graph-ql-type-to-represent-the-custom-data-type)
+
+- [Register the prismic custom type as a routable type](#register-the-prismic-custom-type-as-a-routable-type)
+
+- [Map GraphQL type to component](#map-graph-ql-type-to-component)
+
+- Optionally [add a routable custom type to the site map](#add-a-routable-custom-type-to-the-site-map)
+
+## Create a field to represent the Url of the custom type
+
+To create a custom field to represent the Url of the custom type:
+
+1- Head over to the prismic console and navigate to custom types (https://<prismic_repository_name>.prismic.io/masks/)
+
+2- Select the custom type to make routable
+
+3- Drag and drop "Key Text - Text field for exact match search" from the right to the Main tab on the left.
+
+4- Give the new field a "Field name", "API ID" and optionally a "Field placeholder" (note the "API ID" is an important field and will be used below)
+
+5- Click ok to confirm the inputs
+
+6- Save the changes
+
+**For the sake of simplicity we will assume the name of this field is `url` for the rest of this documentation**
+
+## Create a GraphQL type to represent the custom data type
+
+If no GraphQL type was created a to represent the custom data type its time to do so.
+
+Create a GraphQL type to represent the custom datatype in schema.gql of the server module's directory
+
+```diff
++type Album implements Routable {
++  url: String # the newly added field
++  artist: String
++  title: String
++  release_date: Date
++  path: String # for the routable interface
++}
+```
+
+Note that inorder for the custom type to be routable it needs to implement the core front-commerce [Routable interface](https://gitlab.com/front-commerce/front-commerce/-/blob/2.7.1/src/server/modules/front-commerce/core/schema.gql#L87). Which basically means the GraphQL type will have to have a field called `path` to represent the url of the routable type.
+
+Add a new GraphQL type to the module's resolver:
+
+```diff
+export default {
+  Query: {
+    ...
+  },
++  Album: {
++    path: (content) => {
++      return (
++        "/" + content.url
++      );
++    },
++  },
++};
+```
+
+Note we added a path field resolver that appends a `/` to the `url` field of our custom prismic type.
+
+## Register the prismic custom type as a routable type
+
+To register the custom type as a routeable prismic type it must be registered with the prismic module. First ensure that the `Prismic/core` is added as a module's dependency then use `loaders.Prismic.registerRoutableType` to add the custom type as routable type:
```

Comment:
Typo on routable, `Prismic/Core`, also the first sentence is weird, "To register […]", it must be registered […]"